config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LogoutConfigurer.java

```diff
@@ -470,13 +462,14 @@ private Saml2LogoutResponseValidator logoutResponseValidator() {
 
 		private Saml2LogoutResponseResolver logoutResponseResolver(
 				RelyingPartyRegistrationResolver relyingPartyRegistrationResolver) {
-			if (this.logoutResponseResolver == null) {
-				if (version().startsWith("4")) {
-					return new OpenSaml4LogoutResponseResolver(relyingPartyRegistrationResolver);
-				}
-				return new OpenSaml3LogoutResponseResolver(relyingPartyRegistrationResolver);
+			if (this.logoutResponseResolver != null) {
+				return this.logoutResponseResolver;
 			}
-			return this.logoutResponseResolver;
+			if (ClassUtils.isPresent("org.opensaml.core.xml.persist.impl.PassthroughSourceStrategy",
```

Comment:
It might be more readable to have one place for all the decision logic. Right now there is some here and some in `OpenSaml4LogoutSupportFactory`